src/views/CountryThreeWProjects/Map/index.tsx

```diff
@@ -0,0 +1,399 @@
+import {
+    useState,
+    useMemo,
+    useCallback,
+} from 'react';
+import {
+    _cs,
+    isDefined,
+    unique,
+    listToGroupList,
+} from '@togglecorp/fujs';
+import {
+    useOutletContext,
+} from 'react-router-dom';
+import Map, {
+    MapSource,
+    MapLayer,
+    MapBounds,
+} from '@togglecorp/re-map';
+import getBbox from '@turf/bbox';
+
+import MapPopup from '#components/MapPopup';
+import TextOutput from '#components/TextOutput';
+import MapContainerWithDisclaimer from '#components/MapContainerWithDisclaimer';
+import LegendItem from '#components/LegendItem';
+import { denormalizeList } from '#utils/common';
+import type { GoApiResponse } from '#utils/restRequest';
+import type { CountryOutletContext } from '#utils/outletContext';
+import useGlobalEnums from '#hooks/domain/useGlobalEnums';
+import useTranslation from '#hooks/useTranslation';
+
+import {
+    defaultMapStyle,
+    defaultMapOptions,
+    getPointCirclePaint,
+    getPointCircleHaloPaint,
+    pointColorMap,
+} from '#utils/map';
+import {
+    COLOR_RED,
+    COLOR_BLUE,
+    COLOR_ORANGE,
+    OPERATION_TYPE_EMERGENCY,
+    OPERATION_TYPE_MULTI,
+    OPERATION_TYPE_PROGRAMME,
+} from '#utils/constants';
+
+import i18n from './i18n.json';
+import styles from './styles.module.css';
+
+type District = NonNullable<GoApiResponse<'/api/v2/district/'>['results']>[number];
+type Project = NonNullable<GoApiResponse<'/api/v2/project/'>['results']>[number];
+
+const redPointCirclePaint = getPointCirclePaint(COLOR_RED);
+const bluePointCirclePaint = getPointCirclePaint(COLOR_BLUE);
+const orangePointCirclePaint = getPointCirclePaint(COLOR_ORANGE);
+const sourceOption: mapboxgl.GeoJSONSourceRaw = {
+    type: 'geojson',
+};
+
+interface GeoJsonProps {
+    districtId: number;
+    numProjects: number;
+}
+
+interface ClickedPoint {
+    feature: GeoJSON.Feature<GeoJSON.Point, GeoJsonProps>;
+    lngLat: mapboxgl.LngLatLike;
+}
+
+type ProjectGeoJson = GeoJSON.FeatureCollection<GeoJSON.Point, GeoJsonProps>;
+
+function getOperationType(projectList: Project[]) {
+    const operationTypeList = unique(
+        projectList
+            .filter((d) => isDefined(d.operation_type))
+            .map((d) => ({
+                // FIXME: typings should be fixed in the server
+                id: d.operation_type as number,
+                title: d.operation_type_display,
+            })),
+        (d) => d.id,
+    ) ?? [];
+
+    if (operationTypeList.length === 0) {
+        return undefined;
+    }
+
+    if (operationTypeList.length === 1) {
+        return operationTypeList[0];
+    }
+
+    return {
+        id: OPERATION_TYPE_MULTI,
+        title: 'Multiple types',
+    };
+}
```

Comment:
This is also done in GlobalThreeW/Map
Let's try to re-use this function.

Also, we might need to translate this string.
Let's add a FIXME note for now.

Do we even need to return `title` here?

Reply:
removed the title field